app/utils/client-hints.tsx

```diff
@@ -53,19 +64,20 @@ export function getHints(request?: Request) {
 			// using ignore because it's not an issue with only one hint, but will
 			// be with more than one...
 			// @ts-ignore PR to improve these types is welcome
-			acc[hintName] = hint.transform(
-				getCookieValue(cookieString, hintName) ?? hint.fallback,
-			)
+			acc[hintName] = hint.transform
+				? hint.transform(getCookieValue(cookieString, hintName))
+				: hint.fallback
 			return acc
 		},
 		{} as {
-			[name in ClientHintNames]: ReturnType<
-				(typeof clientHints)[name]['transform']
-			>
+			[name in ClientHintNames]: string
```

Comment:
We lose some type-checking here. But couldn't think of a better way to do it.

Reply:
Ah yeah, we'll need to figure out how to have our cake and eat it too here 😅 If you can't work it out, then that's fine. I'll come back to this later when I find time. Thanks!

Reply:
Couldn't figure out :(